packages/block/src/header.ts

```diff
@@ -389,8 +399,8 @@ export class BlockHeader {
       }
     }
 
-    // Check for constant values for PoS blocks
-    if (this._common.consensusType() === ConsensusType.ProofOfStake) {
+    // Validation for PoS blocks (EIP-3675)
+    if (this._common.isActivatedEIP(3675)) {
```

Comment:
I would like to keep things like this on the `ConsensusType.ProofOfStake` comparison whereever possible. These kind of validations are just Proof-of-Stake dependent, whereever this is coming from (from a consensus change by EIP-3675 or whatever else). This will alow us more flexibility to create e.g. pure PoS blocks from the beginning of a chain.

Logically checking for EIP-3675 is just a narrowing of things, since this by-design also switches to PoS. So directly checking for PoS is the more flexible solution which will allow for more use cases.

I guess the best mental separation is the following: for things which are directly related to the transition itself because there is some kind of trigger on something should get a 3675 check. Things which just define PoS stuff with or without a transition should get a PoS check.

Does this make sense? Or do you have other reasons not directly apparent for this change?

Reply:
yes for sure, didn't think too much about this change, makes sense to keep it as a check on consensusType 👍 will update